labapp/approuter/router.go

```diff
@@ -137,6 +140,29 @@ func (s *router) connect(ctx context.Context, addrs []string) error {
 	return nil
 }
 
+func (s *router) connectOne(ctx context.Context, addrs []string) error {
+	span, ctx := traceutil.StartSpanFromContext(ctx, "approuter.connectOne")
+	defer span.Finish()
+	span.SetTag("addrs", len(addrs))
+
+	// Try to connect to each address in turn, until one of them works
+	var grperr error
+	for _, addr := range addrs {
+		infos, err := parseAddrs([]string{addr})
+		if err != nil {
+			grperr = err
```

Comment:
I think we should fail on `parseAddrs` errors? Are there invalid addrs that we want to ignore?

Reply:
I believe this will fail on parseArrs errors, unless I'm reading it wrong?

Reply:
Sorry, I meant fail fast. If the next `addr` is valid, the previous `grperr` is overwritten.

Reply:
Oh I see, that seems reasonable

Reply:
I'll change